### DISCUSSION TITLE
External DAC over I2S not working

### DISCUSSION THREAD

**Discussion by ld21** [pschatzmann/ESP32-A2DP#377] [Q&A] [answered]
Hello everyone, hi Phil

I like the simplicity of your library and your simple bluetooth receiver works flawlessly if I use the internal DAC. However, if I connect either a MAX98357a [(https://www.adafruit.com/product/3006)](https://www.adafruit.com/product/3006) or a WM8960 Audio Board [(https://www.waveshare.com/w/upload/5/58/WM8960_Audio_Board_User_Manual_en.pdf)](https://www.waveshare.com/w/upload/5/58/WM8960_Audio_Board_User_Manual_en.pdf) with speakers, I get no sound (and no feedback on why). Can anyone recommend an I2S-amplifier that works well with the library? Or give me a hint on how to connect these things (other than indicated in the examples: bck_io_num = 26, ws_io_num = 25,data_out_num = 22).

Thank you very much, Dani

**Comment by pschatzmann** [comment, marked as answer]
I suggest to read the Hardware Wiki from my [Audio Tools](https://github.com/pschatzmann/arduino-audio-tools/wiki) library!
Please note that the A2DP library uses different default pins then the AudioTools

The MAX98357 should be streight forward. You just need to make sure that the hardware pins are matching with what is specified in software!

For the WM8960 you need to setup the codec via a separate driver library. I am providing the WM8960Stream class which does this automatically for you.

Here is the sketch that will work with the output functionality of the AudioTools with the documented default pins in the Wiki: https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-basic-api/basic-a2dp-i2s/basic-a2dp-i2s.ino

**Reply by ld21** [reply]
Great, thank you very much. I wasn't aware that I also have to look at the Audio Tools Library. Now, everything is working fine. Cheers, Dani

**Reply by pschatzmann** [reply]
Well, the MAX98357 should play w/o the AudioTools. 
But there I have much more information
